graphql_compiler/compiler/ir_lowering_match.py

```diff
@@ -445,71 +447,81 @@ def remove_backtrack_blocks_from_fold(folded_ir_blocks):
     return new_folded_ir_blocks
 
 
-def convert_optional_traversals_to_compound_match_query(match_query):
+def _filter_local_field_existence(field_name):
+    """Return an Expression that is True iff `field_name` does not exist."""
+    local_field = LocalField(field_name)
+    local_field_size = UnaryTransformation(u'size', local_field)
+    field_null_check = BinaryComposition(u'=', local_field, NullLiteral)
+    field_size_check = BinaryComposition(u'=', local_field_size, ZeroLiteral)
+    return BinaryComposition(u'||', field_null_check, field_size_check)
+
+
+def _prune_traverse_using_omitted_locations(traverse,
+                                            omitted_locations,
+                                            optional_locations,
+                                            location_to_optional):
+    """Return a prefix of the given traverse, excluding any blocks after an omitted optional."""
+    new_traverse = []
+    for step in traverse:
+        if isinstance(step.root_block, Traverse) and step.root_block.optional:
+            in_optional_location = location_to_optional.get(
+                step.as_block.location, None)
+            if in_optional_location in omitted_locations:
+                # Add filter to indicate that the omitted edge(s) shoud not exist
+                field_name = step.root_block.get_field_name()
+                new_predicate = _filter_local_field_existence(field_name)
+                old_filter = new_traverse[-1].where_block
+                if old_filter:
+                    new_predicate = BinaryComposition(u'&&', old_filter.predicate, new_predicate)
+                new_traverse[-1] = new_traverse[-1]._replace(
+                    where_block=Filter(new_predicate))
+                break
+            elif in_optional_location in optional_locations:
+                new_root_block = Traverse(step.root_block.direction, step.root_block.edge_name)
+                new_traverse.append(step._replace(root_block=new_root_block))
+            else:
+                new_traverse.append(step)
+        else:
+            new_traverse.append(step)
+
+    return new_traverse
+
+
+def convert_optional_traversals_to_compound_match_query(
```

Comment:
Try to improve the docstring. There is very little detail and context as to what is going on here, and some grammar issues in the args.